test/integration/test_simplex.py

```diff
@@ -0,0 +1,58 @@
+import unittest
```

Comment:
Can this be merged with test/unit/test_tag_gmsh.py? Alternatively with unit/test_tag (those two files should also have been merged, but that is another matter).

Reply:
Yes, I didn't see the unit/test_tag_gmsh.py as I was expecting to find it in integration tests. There is a difference between the two. The new one tests the tags before the splitting of the grids, while the unit/test_tag_gmsh.py tests the tags after splitting of the grid. However, I merged the two and moved them to integration tests. 

I also think integration/test_face_tag.py is exactly the same as unit/test_tag.py? I do not think these should be merged with the gmsh stuff because, well, they do not call gmsh. Anyway, I did not touch these files.

Reply:
Well, consolidation of the tests is a major task. I think we'll just go with this. At some point, someone will need to spend two days on sorting out all the tests, and then ideally two weeks deleting tests and writing new ones. Sigh.